pkg/events/derive/symbols_collision.go

```diff
@@ -2,231 +2,329 @@ package derive
-// which has collisions with the newly loaded one.
-// For efficiency reasons, the event uses caching that require the `sched_process_exec` event for handling.
-func SymbolsCollision(
-	soLoader sharedobjs.DynamicSymbolsLoader,
-	symbolsBlackList []string,
-	symbolsWhiteList []string) DeriveFunction {
-	gen := initSOCollisionsEventGenerator(soLoader, symbolsBlackList, symbolsWhiteList)
+//
+// SymbolsCollision --------------------------------------------------------------------------------
+//
+// Generates events for collisions between symbols exported by shared objects loaded to the same
+// process virtual memory address space.
+//
+// One event is created for each symbol collision, and all of them are triggered by the loading of a
+// shared object to a process. For efficiency reasons, the event uses caching that require the
+// `sched_process_exec` event for handling.
+//
+
+func SymbolsCollision(soLoader sharedobjs.DynamicSymbolsLoader, fScopes *filterscope.FilterScopes,
+) DeriveFunction {
+
+	symbolsCollisionFilters := map[string]filters.Filter{}
+
+	// pick white and black lists from the filters (TODO: change this)
+	for fScope := range fScopes.Map() {
+		f := fScope.ArgFilter.GetEventFilters(events.SymbolsCollision)
+		maps.Copy(symbolsCollisionFilters, f)
+	}
+
+	symbolsWhitelist := []string{}
+	symbolsBlacklist := []string{}
+
+	filter, ok := symbolsCollisionFilters["symbols"].(*filters.StringFilter)
+	if filter != nil && ok {
+		symbolsWhitelist = filter.Equal()
+		symbolsBlacklist = filter.NotEqual()
+	}
+
+	gen := initSOCollisionsEventGenerator(soLoader, symbolsBlacklist, symbolsWhitelist)
+
 	return deriveMultipleEvents(events.SymbolsCollision, gen.deriveArgs)
 }
 
-// SymbolsCollisionArgsGenerator is the struct responsible to create the SO symbols collisions derived event.
-// To do so, it uses multiple caches to accelerate performance and reduce chances for failure.
+//
+// SymbolsCollisionArgsGenerator -------------------------------------------------------------------
+//
+
+// SymbolsCollisionArgsGenerator creates the shared object symbols collisions derived events. To do
+// so, it uses multiple caches to accelerate performance and reduce chances for failure.
 type SymbolsCollisionArgsGenerator struct {
-	soLoader          sharedobjs.DynamicSymbolsLoader
-	processesSOsCache processesLoadedObjectsCache
-	collisionsCache   sharedObjectsCollisionsCache
-	symbolsBlackList  map[string]bool
-	symbolsWhiteList  map[string]bool
-	returnedErrors    map[string]bool
+	soLoader               sharedobjs.DynamicSymbolsLoader
+	loadedObjsPerProcCache loadedObjsPerProcessCache // cache of loaded shared objects per process
+	collisionChecksCache   collisionChecksCache      // cache of symbols collisions
+	symbolsBlacklistMap    map[string]bool           // will delete symbols NOT in this map
+	symbolsWhitelistMap    map[string]bool           // will delete symbols IN this map
```

Comment:
```suggestion
	deniedSymbolsMap    map[string]bool           // will delete symbols NOT in this map
	allowedSymbolsMap   map[string]bool           // will delete symbols IN this map
```

And for all related.